### DISCUSSION TITLE
v2.7.0 Released

### DISCUSSION THREAD

**Discussion by iBotPeaches** [iBotPeaches/Apktool#2954] [Releases]
## What's Changed
* feat: introduce SDK_S_V2 by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2780
* Optimize CI a bit by @Goooler in https://github.com/iBotPeaches/Apktool/pull/2773
* fix: properly release resource by @auermich93 in https://github.com/iBotPeaches/Apktool/pull/2783
* feat: update internal framework to API 32 (12L Release) by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2743
* Revert "fix: decoding references to private resources" by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2785
* New aapt/aapt2 binaries by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2787
* test (aapt2): add assertion for "usesPermissionFlags" by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2790
* test (aapt/aapt2): ensure ampersand is escaped on android:label by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2802
* added option to include generic/permissive network security config file durin… by @erev0s in https://github.com/iBotPeaches/Apktool/pull/2791
* fix: redesign StyledString decoding by @IgorEisberg in https://github.com/iBotPeaches/Apktool/pull/2816
* Tests / Style for #2815 by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2817
* build: cancel duplicate workflows by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2818
* feat: update internal framework to API 33 (Tiramisu Privacy Preview) by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2845
* Binaries: aapt/aapt2 updates (July 5, 2022) by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2846
* Update apktool.bat by @jonathanschoeller in https://github.com/iBotPeaches/Apktool/pull/2882
* Fixes facebook apk deconstruction by @tvildo in https://github.com/iBotPeaches/Apktool/pull/2884
* Support malformed resource names/namespaces by falling back to resId by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2886
* Support Spare Flags by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2887
* build: migrate codeql to different workflow by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2888
* fix: correct npe if null is returned from attribute value by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2889
* Fixes #2900 by @MaSven in https://github.com/iBotPeaches/Apktool/pull/2901
* fix: overlay spec parsing (#2805) by @gramound in https://github.com/iBotPeaches/Apktool/pull/2917
* Make default framework location detection on Linux XDG compliant by @Miepee in https://github.com/iBotPeaches/Apktool/pull/2924
* Fix « doNotCompress » in case of obfuscated resources. by @Kirlif in https://github.com/iBotPeaches/Apktool/pull/2925
* fix: don't assume a referent exists on ReferenceValue by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2926
* Update apktool.bat by @Build-0-Matic in https://github.com/iBotPeaches/Apktool/pull/2930
* Revert "Update apktool.bat" by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2931
* fix: support properly mapping r/R/res resources during disassemble by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2936
* fix:  res/layout/(name removed).xml: Invalid file name by @Danealau in https://github.com/iBotPeaches/Apktool/pull/2940
* build: update to non-vulnerable commons_text/snakeyaml by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2942
* Issue 2943 - Support raw disassembly w/ AndResGuard by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2944
* build: include smali from source by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2941
* build: filter jitpack to only com.github.ibotpeaches.* by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2953
* build: support jdk17 by @iBotPeaches in https://github.com/iBotPeaches/Apktool/pull/2952

## New Contributors
* @auermich93 made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2783
* @erev0s made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2791
* @jonathanschoeller made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2882
* @tvildo made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2884
* @MaSven made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2901
* @gramound made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2917
* @Miepee made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2924
* @Kirlif made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2925
* @Build-0-Matic made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2930
* @Danealau made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2940
* @Surendrajat made their first contribution in https://github.com/iBotPeaches/Apktool/pull/2927

**Full Changelog**: https://github.com/iBotPeaches/Apktool/compare/v2.6.1...v2.7.0
**Release Post**: https://connortumbleson.com/2022/11/24/apktool-v2-7-0-released

<hr /><em>This discussion was created from the release <a href='https://github.com/iBotPeaches/Apktool/releases/tag/v2.7.0'>v2.7.0 Released</a>.</em>

**Comment by surendrajat** [comment]
Oops, my pr missing 😆

**Reply by iBotPeaches** [reply]
fixed. I assume this happened because I closed yours and remade it.

**Comment by RenateUSB** [comment]
I can confirm that the hidden API data is correct now (15,000 classes).
A round-trip on framework.jar still breaks my Android. Investigating further.
